arch/riscv/Kconfig

```diff
@@ -90,6 +90,12 @@ config RISCV_SOC_INTERRUPT_INIT
 	  Enable SOC-based interrupt initialization
 	  (call soc_interrupt_init, within _IntLibInit when enabled)
 
+config RISCV_SOC_INIT_GP_VALUE
+	bool "Enable SOC-based global pointer register initialization"
+	help
+	  Enable SOC-based pointer register initialization
+	  (call __soc_get_gp_initial_value when initializing a thread)
+
```

Comment:
I think this is not SoC support. It's better to separate another PR or commit.

Reply:
Thanks @katsuster I split this section in a separate commit.